src/libraries/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -222,116 +210,125 @@ public void SendAsync_HttpTracingEnabled_Succeeds(bool useSsl)
             }, UseVersion.ToString(), useSsl.ToString()).Dispose();
         }
 
-        [OuterLoop("Uses external servers")]
         [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
         public void SendAsync_ExpectedDiagnosticExceptionLogging()
         {
-            RemoteExecutor.Invoke(useVersionString =>
+            RemoteExecutor.Invoke(async (useVersion, testAsync) =>
             {
-                bool exceptionLogged = false;
-                bool responseLogged = false;
+                Exception exceptionLogged = null;
+
+                SemaphoreSlim responseLoggedSemaphore = new(0, 1);
+
                 var diagnosticListenerObserver = new FakeDiagnosticListenerObserver(kvp =>
                 {
                     if (kvp.Key.Equals("System.Net.Http.Response"))
                     {
                         Assert.NotNull(kvp.Value);
-                        var requestStatus =
-                            GetPropertyValueFromAnonymousTypeInstance<TaskStatus>(kvp.Value, "RequestTaskStatus");
+                        TaskStatus requestStatus = GetProperty<TaskStatus>(kvp.Value, "RequestTaskStatus");
                         Assert.Equal(TaskStatus.Faulted, requestStatus);
-
-                        responseLogged = true;
+                        responseLoggedSemaphore.Release();
                     }
                     else if (kvp.Key.Equals("System.Net.Http.Exception"))
                     {
                         Assert.NotNull(kvp.Value);
-                        GetPropertyValueFromAnonymousTypeInstance<Exception>(kvp.Value, "Exception");
-
-                        exceptionLogged = true;
+                        exceptionLogged = GetProperty<Exception>(kvp.Value, "Exception");
                     }
                 });
 
                 using (DiagnosticListener.AllListeners.Subscribe(diagnosticListenerObserver))
                 {
-                    diagnosticListenerObserver.Enable(s => !s.Contains("HttpRequestOut"));
-                    using (HttpClient client = CreateHttpClient(useVersionString))
-                    {
-                        Assert.ThrowsAsync<HttpRequestException>(() => client.GetAsync($"http://_{Guid.NewGuid().ToString("N")}.com"))
-                            .GetAwaiter().GetResult();
-                    }
+                    diagnosticListenerObserver.Enable();
+
+                    var invalidUri = new Uri($"http://_{Guid.NewGuid().ToString("N")}.com");
```

Comment:
nit: rather than generating a GUID-based Uri, we can use a Uri that ends in ".invalid" (like "nosuchhost.invalid", which I think we use elsewhere) since this is guaranteed to not resolve. 

https://en.wikipedia.org/wiki/.invalid

Reply:
Much cleaner than GUIDs everywhere 👍
Changed